maven/bnd-export-maven-plugin/src/main/java/aQute/bnd/maven/export/plugin/ExportMojo.java

```diff
@@ -99,7 +103,6 @@ private void export(Run run) throws Exception {
 	private void resolve(StandaloneRun run) throws Exception, IOException {
 		try (ProjectResolver projectResolver = new ProjectResolver(run)) {
 			List<Container> runBundles = projectResolver.getRunBundles();
```

Comment:
It is not about groovy and closures. It is about using the keyset of the resolve result vs. calling getRunBundles which converts the Resources to Container so that you have to convert them back to Resources. Just do like the gradle plugin does and use the keyset of the resolve result. Also when you sort, you need to sort by bsn and then by version (as the gradle plugin does.)